gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/JobStatus.java

```diff
@@ -0,0 +1,110 @@
+package com.google.gcloud.bigquery;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google BigQuery Job status. Objects of this class can be examined when polling an asynchronous
+ * job to see if the job completed.
+ */
+public class JobStatus implements Serializable {
+
+  private static final long serialVersionUID = -714976456815445365L;
+  public enum State {
```

Comment:
Nit, empty line above plus a javadoc would be nice (it may also be good to explain that failed job's status is also DONE but with a non null error).